scheduler/src/cook/mesos/quota.clj

```diff
@@ -27,65 +28,67 @@
                 [?e :quota/user ?u]
                 [?e :quota/resource ?r]
                 [?r :resource/type ?t]
-                [?r :resource/amount ?a]]]
-    (or (-> (q query db user type) ffirst)
-        (-> (q query db default-user type) ffirst)
-        (Double/MAX_VALUE))))
+                [?r :resource/amount ?a]
+                [(cook.mesos.pool/check-pool $ ?r :resource/pool ?pool-name
+                                             ?requesting-default-pool)]]]
+    (ffirst (q query db user type pool-name' requesting-default-pool))))
 
-(defn- get-max-jobs-quota
-  [db user]
-  [:count
-   (or (:quota/count (d/entity db [:quota/user user]))
-       (:quota/count (d/entity db [:quota/user default-user]))
-       (Double/MAX_VALUE))])
+(defn- get-quota-by-type-or-default
+  [db type user pool-name]
+  (or (get-quota-by-type db type user pool-name)
+      (get-quota-by-type db type default-user pool-name)
+      Double/MAX_VALUE))
 
 (defn get-quota
   "Query a user's pre-defined quota.
 
    If a user's pre-defined quota is NOT defined, return the quota for the
    `default-user`. If there is NO `default-user` value for a specific type,
    return Double.MAX_VALUE."
-  [db user]
-  (->> (util/get-all-resource-types db)
-       (map (fn [type] [type (get-quota-by-type db type user)]))
-       (cons (get-max-jobs-quota db user))
-       (into {})))
-
-(defn quota-history
-  "Return changes to a user's own quota, in the form
-  [{:time (inst when change occurred)
-    :reason (stated reason for change)
-    :quota (quota as would have been returned by get-quota after change)}]"
-  [db user]
-  (->> (d/q '[:find ?e ?a ?v ?added ?tx
-              :in $ ?a ?e
-              :where
-              [?e ?a ?v ?tx ?added]]
-            (d/history db)
-            :quota/reason
-            [:quota/user user])
-       (sort-by last)
-       (map (fn [[e attr v added tx]]
-              (let [tx-db (d/as-of db tx)
-                    tx-e (d/entity tx-db e)]
-                {:time (:db/txInstant (d/entity db tx))
-                 :reason (:quota/reason tx-e)
-                 :quota (get-quota tx-db user)})))
-       distinct))
+  [db user pool-name]
+  (let [mesos-resource-quota (->> (util/get-all-resource-types db)
+                                  (map (fn [type] [type (get-quota-by-type-or-default db type user pool-name)])))
+        count-quota (or (get-quota-by-type db :resource.type/count user pool-name) ; prefer resource
+                        (:quota/count (d/entity db [:quota/user user])) ; then the field on the user
+                        (get-quota-by-type db :resource.type/count default-user pool-name) ; then the resource from the default user
+                        (:quota/count (d/entity db [:quota/user default-user])) ; then the field on the default user
+                        Double/MAX_VALUE)]
+    (->> mesos-resource-quota
+         (cons [:count (double count-quota)])
+         (into {}))))
```

Comment:
I don't like it in the docstring, since it seems like an implementation detail for the function but I'll add a comment to the top of the namespace explaining it.